pkg/binlog/event/event.go

```diff
@@ -851,3 +851,19 @@ func GenDummyEvent(header *replication.EventHeader, latestPos uint32, eventSize
 	ev, err := GenQueryEvent(&headerClone, latestPos, 0, 0, 0, nil, nil, queryBytes)
 	return ev, err
 }
+
+// GenHeartbeatEvent generates a heartbeat event.
+// ref: https://dev.mysql.com/doc/internals/en/heartbeat-event.html
+func GenHeartbeatEvent(header *replication.EventHeader) *replication.BinlogEvent {
+	// modify header
+	headerClone := *header // do a copy
+	headerClone.Flags = 0
+	headerClone.EventSize = 39
```

Comment:
does MySQL choose not keep size? I think this may cause positions in `query-status` mismatched which may confuse user

Reply:
It keeps the LogPos but not EventSize.

Reply:
The meets problem when we calculate startPosition, but we don't calculate it for heartbeat event.